compiler/rustc_resolve/src/late.rs

```diff
@@ -193,6 +193,27 @@ pub(crate) enum RibKind<'ra> {
     /// No restriction needs to be applied.
     Normal,
 
+    /// During resolving an item in a block, we had records all bindings defined
+    /// in this local scope, for these features:
+    ///
+    /// - Forward reference detection:
+    ///
+    /// ```ignore (illustrative)
+    /// let a = b;  // displays '`b` is defined at <span>' instead of ''b' not found'
```

Comment:
I haven't seen this in tests.

Reply:
Yes, there is an item that requires optimization, the reason this hasn't been implemented yet is that I consider it to be independent work. Similar issues can be found in:

https://github.com/rust-lang/rust/pull/143141/files#diff-f6b11904658a0fbaf9bc5c81106db27d91e9b084f78da6978804cf2fd303fc5eR24